### PR TITLE
#341 2.4.6 hidden events

### PR DESCRIPTION
**Pull request by nilsthamm**
ref #341

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -7,11 +7,20 @@ class EventsController < ApplicationController
 
   # GET /events
   def index
-    @events = Event.sorted_by_start_date(!can?(:view_unpublished, Event))
+    @events = Event.sorted_by_start_date(false)
+    if !can? :view_hidden, Event
+      @events = @events.select{|event| event.hidden == false}
+    end
+    if !can? :view_unpublished, Event
```
Comment (cmfcmf) [inline app/controllers/events_controller.rb]: This is unrelated to the PR, right? I assume this was a bug?

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -7,11 +7,20 @@ class EventsController < ApplicationController
 
   # GET /events
   def index
-    @events = Event.sorted_by_start_date(!can?(:view_unpublished, Event))
+    @events = Event.sorted_by_start_date(false)
+    if !can? :view_hidden, Event
```
Comment (cmfcmf) [inline app/controllers/events_controller.rb]: nit: You could use `unless`.

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (cmfcmf) [reply app/controllers/events_controller.rb]: nit: This could also benefit from `unless`.

File: app/views/events/_form.html.erb [hunk 3]
```diff
@@ -13,7 +13,13 @@
       <%=f.error_span(:kind) %>
     </div>
   </div>
-
+  <div class="form-group">
+    <div class="col-lg-offset-2 col-lg-10">
+      <%= f.check_box :hidden%>
```
Comment (cmfcmf) [inline app/views/events/_form.html.erb]: nit: missing space before `%>`

File: app/views/events/_form.html.erb [hunk 4]
```diff
@@ -13,7 +13,13 @@
...
+      <%= f.label :hidden, :class => 'control-label' %>
+      <%=f.error_span(:hidden) %>
```
Comment (cmfcmf) [inline app/views/events/_form.html.erb]: nit: missing space after `<%=`

File: spec/controllers/events_controller_spec.rb [hunk 5]
```diff
@@ -165,6 +165,17 @@
     end
   end
 
+  describe "GET #show for hidden event as pupil"
+    it "should redirect to new application letter page" do
+      @event = FactoryGirl.create(:event, hidden: true)
+      @user = FactoryGirl.create(:user, role: :pupil)
+      @user.profile ||= FactoryGirl.create(:profile)
```
Comment (cmfcmf) [inline spec/controllers/events_controller_spec.rb]: Use `:user_with_profile` instead

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (cmfcmf) [reply app/controllers/events_controller.rb]: > This is unrelated to the PR, right? I assume this was a bug?

Also, I don't see a new test for this feature.

**Review by cmfcmf — CHANGES_REQUESTED** [review]
Mostly minor nitpicks, looks good overall!

File: app/controllers/events_controller.rb (hunk 1 above)
Comment (nilsthamm) [reply app/controllers/events_controller.rb]: > This is unrelated to the PR, right? I assume this was a bug?

the unpuplished events were filtered before by the sorted_by_start_date function (so this is already tested in spec/features/event_spec.rb:28). 
Because i wanted to distinguish between hidden and unpuplished events, I test the rigths to view them separately. I know at the moment, this distinction makes no difference, but i don't think it is a good idea to summerize it to one ability.

**Review by cmfcmf — APPROVED** [review]
LGTM now!
